packages/sitecore-jss/src/utils/resolve-url.ts

```diff
@@ -0,0 +1,50 @@
+import isServer from './is-server';
+import querystring from 'querystring';
+
+/**
+ * note: encodeURIComponent is available via browser (window) or natively in node.js
+ * if you use another js engine for server-side rendering you may not have native encodeURIComponent
+ * and would then need to install a package for that functionality
+ * @param {Object} params
+ */
+function getQueryString(params: querystring.ParsedUrlQueryInput) {
+  return Object.keys(params)
+    .map((k) => `${encodeURIComponent(k)}=${encodeURIComponent(String(params[k]))}`)
+    .join('&');
+}
+
+/**
+ * Resolves a base URL that may contain query string parameters and an additional set of query
+ * string parameters into a unified string representation.
+ * @param {string} urlBase the base URL that may contain query string parameters
+ * @param {Object} params query string parameters
+ * @returns a URL string
+ * @throws {RangeError} if the provided url is an empty string
+ */
+function resolveUrl(urlBase: string, params: querystring.ParsedUrlQueryInput = {}): string {
+  if (!urlBase) {
+    throw new RangeError('url must be a non-empty string');
+  }
+
+  // This is a better way to work with URLs since it handles different user input
+  // edge cases. This works in Node and all browser except IE11.
+  // TODO: Verify our browser support requirements.
+  if (isServer()) {
+    const url = new URL(urlBase);
+    for (const key in params) {
+      if ({}.hasOwnProperty.call(params, key)) {
```

Comment:
Can we simplify to `params.hasOwnProperty(key)`? And if `const key in params` executed, then the `key` exists in `params`?

Reply:
That's how I wrote it at first, but this is how lint wants it (https://eslint.org/docs/rules/no-prototype-builtins)